### ISSUE TITLE
LibreOffice Writer not showing in Dock

### ISSUE THREAD

**Issue by KodiakIT** [robint99/mate-dock-applet#139]
As stated in the title, it would seem that LibreOffice Writer does not properly display in the dock when a window is opened. Other LibreOffice applications seem to display properly, other window switchers seem to recognize it, so it seems to be an issue specifically between the dock applet and LOW.

**Comment by robint99** [comment]
All seems to be working for me. What version of the applet and what distribution are you using?

**Comment by gabrc52** [comment]
Confirming this issue, some icons don't appear until running  `pkill dock_applet`
(V0.85)

**Comment by robint99** [comment]
Again, all seems to be ok for me on 18.04. Could you just confirm, is this just a LibreOffice problem, or are there any other icons that don't appear when they should?

**Comment by gabrc52** [comment]
It's not just LibreOffice, but I've been able to reproduce it more times with LibreOffice. This could also be due to Ubuntu MATE not shipping an up-to-date version of the applet.

**Comment by CreamyCookie** [comment]
@gabrc52 But I've build mate-dock-applet from master and still had the problem.

**Comment by robint99** [comment]
I've finally been able to reproduce this problem (by repeatedly opening and Writer, Calc and Impress and closing them as quickly as possible). Both Calc and Writer would occasionally not appear in the dock when there they were opened. Also, I'd sometimes get a generic cog icon which would appear and then not go away. 

I've just committed a fix which solves this, for me at least. Hopefully, it'll also fix it for everyone else who had problems with icons not appearing etc.
